llvm/test/Transforms/ConstantHoisting/RISCV/immediates.ll

```diff
@@ -211,11 +211,12 @@ exit:
 
 ; Check that we use a common base for immediates needed by a store if the
 ; constants require more than 1 instruction.
-; TODO: This doesn't trigger currently.
 define void @test20(ptr %p1, ptr %p2) {
 ; CHECK-LABEL: test20
-; CHECK: store i32 15111111, ptr %p1
-; CHECK: store i32 15111112, ptr %p2
+; CHECK: %const = bitcast i32 15111111 to i32
```

Comment:
Please add test cases for:
* Imm=0 value with legal type
* Imm=0 value with illegal type
* Partially zero value with illegal type
* Misaligned case